src/kmod/hash-table-generic.h

```diff
@@ -23,23 +23,31 @@
 //
 
 struct HashTbl {
-	struct hlist_head *tablePtr;
-	struct list_head   genTables;
-	uint64_t	   numberOfBuckets;
-	uint64_t	   tableSpinlock;
-	atomic64_t	   tableInstance;
-	atomic64_t	   tableAllocs;
-	atomic64_t	   tableShutdown; // shutting down = 1 running = 0
-	int		   key_len;
-	struct kmem_cache *hash_cache;
-	uint8_t		   name[20];
-	int		   key_offset;
-	int		   node_offset;
-	bool		   base_size;
+	struct hashtbl_bkt *tablePtr;
+	struct list_head    genTables;
+	uint64_t	    numberOfBuckets;
+	uint32_t	    secret;
+	atomic64_t	    tableInstance;
+	atomic64_t	    tableAllocs;
+	atomic64_t	    tableShutdown; // shutting down = 1 running = 0
+	int		    key_len;
+	struct kmem_cache * hash_cache;
+	uint8_t		    name[20];
```

Comment:
Please #define the number 20. There is strncpy which incorrectly uses sizeof(hashTblp->name) which should be a byte less.

Reply:
Yeah I agree. This a copy n paste.